### PR TITLE
Pill icons based on master reagent and transparent chem bottles

### PR DESCRIPTION
**Pull request by PeterArbeitsfrei**
Having all pills made my chemistry look the same is boring and confusing.

This PR adds different pill icons for pills which master reagent is a toxin, drug or medicine.

I also changed some standard pill icons to make them not all look the same / look like toxins.

Also I added a transparent bottle type so chemistry isn't as shitty as before with 20 random choosen icons... of course the old bottles remain and will continued to be used for mapspawned bottles, so this changes only affects chemists. (for now)

transparent_bottles

:cl: Bobylein
rscadd: Nanotrasen is finally able to source transparent bottles for chemistry.
/:cl:

### CODE REVIEW

**Comment by xxalpha** [comment]
Some example pictures please?

**Comment by Bobylein** [comment]
oops, yea added above

**Comment by phil235** [comment]
is the empty space inside the bottle  still clickable?

**Comment by Bobylein** [comment]
@phil235 yea

File: code/modules/reagents/reagent_containers/pill.dm [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	name = "pill"
 	desc = "A tablet or capsule."
 	icon = 'icons/obj/chemical.dmi'
-	icon_state = "pill"
+	//icon_state = "pill"
```
Comment (phil235) [inline code/modules/reagents/reagent_containers/pill.dm]: if you're going to remove it, remove the line entirely, don't just comment it.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 2]
```diff
@@ -3,15 +3,59 @@
...
+/obj/item/weapon/reagent_containers/glass/bottle/on_reagent_change()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/bottle/pickup(mob/user)
+	..()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/bottle/dropped(mob/user)
+	..()
+	update_icon()
```
Comment (phil235) [inline code/modules/reagents/reagent_containers/bottle.dm]: Does the update on pickup and drop really do anything?

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 3]
```diff
@@ -3,15 +3,59 @@
...
+
+/obj/item/weapon/reagent_containers/glass/bottle/attack_hand()
+	..()
+	update_icon()
```
Comment (phil235) [inline code/modules/reagents/reagent_containers/bottle.dm]: same here

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 4]
```diff
@@ -3,15 +3,59 @@
...
+
+/obj/item/weapon/reagent_containers/glass/bottle/update_icon()
+	overlays.Cut()
+
+
+
+
```
Comment (phil235) [inline code/modules/reagents/reagent_containers/bottle.dm]: why the huge empty space here?

**Comment by lordpidey** [comment]
I'm against this change, traitor chemists don't need a giant "THIS PILL IS ACTUALLY POISON" sign.

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 2 above)
Comment (Bobylein) [inline code/modules/reagents/reagent_containers/bottle.dm]: I just took it from the beakers, will test it.

Does throwing items count as dropped?

**Comment by Bobylein** [comment]
@lordpidey you mean they actually need to make some effort? like mixing different poisons and for example plus some added Perfluor to make it look like a proper medical pill?

Chemist is by far my favorite role and I played it as antag many times, handing out poison pills en masse  is not something one does anyway as security/cmo will bust your ass for it.

But I agree that it's not perfect, that's why I planned to add a button to the chemmaster so you can choose if you want all pills to look the same or not.
(actually, the plan was that you chould choose whatever sprite you want but tgui doesn't like dmi files...)

**Comment by Bawhoppen** [comment]
this does not need a changelog

**Comment by Bobylein** [comment]
@Bawhoppen I would disagree, as there are changelog for things like 
"Legion's souls are now more obviously inert when becoming inert."
But if you are worried that it takes too much space, I can of course remove them.

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 4 above)
Comment (Bobylein) [inline code/modules/reagents/reagent_containers/bottle.dm]: gone

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 3 above)
Comment (Bobylein) [inline code/modules/reagents/reagent_containers/bottle.dm]: what same? attack_hand()?

Comment (phil235) [inline code/modules/reagents/reagent_containers/bottle.dm]: you don't need to call update_icon() on attack_hand().

**Comment by Bawhoppen** [comment]
you think I didn't try to shoot down those changelogs too? changelogs for only important things people won't figure out on their own

**Comment by xxalpha** [comment]
If it is relevant to the players, you make a changelog for it. This is relevant to the players.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 5]
```diff
@@ -3,15 +3,59 @@
...
+/obj/item/weapon/reagent_containers/glass/bottle/dropped(mob/user)
+	..()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/bottle/attack_hand()
+	..()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/bottle/update_icon()
+	overlays.Cut()
```
Comment (Razharas) [inline code/modules/reagents/reagent_containers/bottle.dm]: :sob: :cry:  :crying_cat_face: :cry: :sob:

**Comment by Bawhoppen** [comment]
No, you shouldn't make changelogs for every single tiny thing

**Comment by LatD** [comment]
I like the bottle change, that will help chemists. But the pill change is a rather big nerf for traitors who want to make fake pills. Chemists use bottles themselves so it's good that they can see what's inside without examining but pills are used differently and they're for other people than chemists.

And the changelog says "Nanotrassen"

**Comment by Bobylein** [comment]
@LatD ah damn, I thought it was written that way, that's my german coming through :P

About the pills, well it's not as chemists couldn't make fake pills and I don't like that every fucking pill look the same right now, but the pure random function before wasn't good either, so I will add a button to choose if you want default pill or not.

**Comment by Iamgoofball** [comment]
boo hoo because players are totally going to recognize and memorize all chemical colors to tell whats in a pill

stop overreacting faggots

**Comment by PKPenguin321** [comment]
It'd be less of an issue if medicines had more colors than pinks and occasional blues

**Comment by RandomMarine** [comment]
>Every other traitor just gets a holoeswordwhatever and murderbones
>Let's make other traitoring methods take more effort.

**Comment by Supermichael777** [comment]
remember that a shit-ton of our medicines are actually dangerous if swallowed. Styptic powder isn't good for your gut. just make a brute pill with a side of cyanide

**Comment by Bawhoppen** [comment]
bar drinks are the most dangerous chems now

**Comment by pudl** [comment]
Please remove the pill change and keep the bottles. I love the color bottles, i tried to do it myself but i couldnt get it to work

**Comment by Iamgoofball** [comment]
the pill change is fine, ignore baw/pidey

**Comment by ChangelingRain** [comment]
I don't like the bottles, they don't look as good as the bottles we have right now.

**Comment by xxalpha** [comment]
It's the same sprite but with different colors.

**Comment by ChangelingRain** [comment]
No, it's not??? It's lacking a lot of the shading the bottle it's based on had.

**Comment by Iamgoofball** [comment]
@ChangelingRain easy fix tbh

**Comment by pudl** [comment]
@Iamgoofball The pills arent colored like the bottles, have have 3 different sprites that change depending on if the drug is a toxin, drug or medicine. So its immediately obvious what kind of drug youre taking

**Comment by Iamgoofball** [comment]
ohhhhh, fuck me
@Bobylein Can you make the pills just greyscale and recolor based on the reagent color mix ala the bottles?

**Comment by lordpidey** [comment]
If pills recolored based on reagent color, I think that would be great.  But right now a giant green "THIS ARE POISON!" marker on pills seems dumb.

**Comment by Bawhoppen** [comment]
wait goof what do you mean ignore me I am neutral I just don't want a changelog for it

**Comment by Iamgoofball** [comment]
sorry, habit

**Comment by Fox-McCloud** [comment]
:+1: for bottles

:-1: for pills

**Comment by Cheridan** [comment]
Why is it based on master reagent instead of all the reagents blended together? Isn't that what beakers do?

**Comment by Bobylein** [comment]
@Cheridan Actually you are right, and it uses the blend color >.<

@Iamgoofball That's what I planned as a next update anyway but thought of it more like a custom option.

@ChangelingRain Yea I tried my best to replicate the shading but I am no spriter.

**Comment by Bobylein** [comment]
@Iamgoofball so tried to color the pills, absolutely not happy with the results :/

screenshot 2016-05-28 16 18 58

**Comment by Iamgoofball** [comment]
there's a PR up that makes the chem colors better

Did you make the greyscale sprite bright or dark?

**Comment by Bobylein** [comment]
@Iamgoofball bright, the biggest problem what I see is the shading with dark colors.

**Comment by Iamgoofball** [comment]
upload a picture of the greyscaled sprite

**Comment by Bobylein** [comment]
reagentfillings

**Comment by Iamgoofball** [comment]
huh, I wonder why it's turning out so dark

my suggestion is to keep going with greyscale pills and wait for #17981 to get merged, it makes stuff a ton more colorful

**Comment by Razharas** [comment]
@Bobylein 
The border of the pills need to change hue too otherwise it looks strange

**Comment by Iamgoofball** [comment]
yeah what raz said

**Comment by optimumtact** [comment]
Ready ?

**Comment by Bobylein** [comment]
No, the pills still look like shit. On the other hand I can take out the pill stuff for now as I don't get it to look good.

**Comment by optimumtact** [comment]
If you think it's best then move the pill changes to another pr and I'll merge the bottle changes.

**Comment by Bobylein** [comment]
Okay did that, the remaining pill changes I would like to leave there to make the map spawned pills more distinctive like they were before goonchem

**Comment by optimumtact** [comment]
Thanks

**Pull request merged**